code/modules/dwarven/dwarven_structures.dm

```diff
@@ -0,0 +1,145 @@
+/obj/structure/destructible/dwarven
+	density = TRUE
+	anchored = TRUE
+	icon = 'icons/obj/dwarven.dmi'
+	light_power = 1
+	var/cooldowntime = 0
+	//break_sound = 'sound/hallucinations/veryfar_noise.ogg' //to do: find a suitable noise
+	//debris = list(/obj/item/stack/sheet/runed_metal = 1) //to do : make a dwarven metal datum
+
+/obj/structure/destructible/dwarven/New()
+	START_PROCESSING(SSfastprocess, src)
+	..()
+
+/obj/structure/destructible/dwarven/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+
+/obj/structure/destructible/dwarven/dwarven_sarcophagus
+	name = "Ancient Sarcophagus"
+	icon_state = "sarcophagus_open"
+
+	var/recharge = TRUE
+	var/recharge_points = 0
+	var/recharge_points_max = 500
+
+/obj/structure/destructible/dwarven/dwarven_sarcophagus/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>It requires [recharge_points_max-recharge_points] points to reactivate </span>"
```

Comment:
This explanation seems arbitrary, maybe instead show a percentage and says its x% charged or something similar